compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -10,25 +10,216 @@ module Language.Bond.Codegen.Java.Class_java
+structFieldDescriptorInitTypeExpr java type_ = javaStructFieldDescriptorInitTypeExpr type_
+    where
+        javaStructFieldDescriptorInitTypeExpr (BT_Maybe t) = javaStructFieldDescriptorInitTypeExpr t
+        javaStructFieldDescriptorInitTypeExpr BT_Int8 = [lt|com.microsoft.bond.BondTypes.INT8|]
+        javaStructFieldDescriptorInitTypeExpr BT_Int16 = [lt|com.microsoft.bond.BondTypes.INT16|]
+        javaStructFieldDescriptorInitTypeExpr BT_Int32 = [lt|com.microsoft.bond.BondTypes.INT32|]
+        javaStructFieldDescriptorInitTypeExpr BT_Int64 = [lt|com.microsoft.bond.BondTypes.INT64|]
+        javaStructFieldDescriptorInitTypeExpr BT_UInt8 = [lt|com.microsoft.bond.BondTypes.UINT8|]
+        javaStructFieldDescriptorInitTypeExpr BT_UInt16 = [lt|com.microsoft.bond.BondTypes.UINT16|]
+        javaStructFieldDescriptorInitTypeExpr BT_UInt32 = [lt|com.microsoft.bond.BondTypes.UINT32|]
+        javaStructFieldDescriptorInitTypeExpr BT_UInt64 = [lt|com.microsoft.bond.BondTypes.UINT64|]
+        javaStructFieldDescriptorInitTypeExpr BT_Float = [lt|com.microsoft.bond.BondTypes.FLOAT|]
+        javaStructFieldDescriptorInitTypeExpr BT_Double = [lt|com.microsoft.bond.BondTypes.DOUBLE|]
+        javaStructFieldDescriptorInitTypeExpr BT_Bool = [lt|com.microsoft.bond.BondTypes.BOOL|]
+        javaStructFieldDescriptorInitTypeExpr BT_String = [lt|com.microsoft.bond.BondTypes.STRING|]
+        javaStructFieldDescriptorInitTypeExpr BT_WString = [lt|com.microsoft.bond.BondTypes.WSTRING|]
+        javaStructFieldDescriptorInitTypeExpr BT_Blob = [lt|com.microsoft.bond.BondTypes.BLOB|]
+        javaStructFieldDescriptorInitTypeExpr (BT_Bonded t) = [lt|bondedOf(#{javaStructFieldDescriptorInitTypeExpr t})|]
+        javaStructFieldDescriptorInitTypeExpr (BT_Nullable t) = [lt|nullableOf(#{javaStructFieldDescriptorInitTypeExpr t})|]
+        javaStructFieldDescriptorInitTypeExpr (BT_Vector t) = [lt|vectorOf(#{javaStructFieldDescriptorInitTypeExpr t})|]
+        javaStructFieldDescriptorInitTypeExpr (BT_List t) = [lt|listOf(#{javaStructFieldDescriptorInitTypeExpr t})|]
+        javaStructFieldDescriptorInitTypeExpr (BT_Set t) = [lt|setOf(#{javaStructFieldDescriptorInitTypeExpr t})|]
+        javaStructFieldDescriptorInitTypeExpr (BT_Map k v) = [lt|mapOf(#{javaStructFieldDescriptorInitTypeExpr k}, #{javaStructFieldDescriptorInitTypeExpr v})|]
+        javaStructFieldDescriptorInitTypeExpr (BT_TypeParam param) = [lt|#{paramName param}|]
+        javaStructFieldDescriptorInitTypeExpr (BT_UserDefined e@Enum {..} _) = [lt|#{qualifiedDeclaredTypeName java e}.BOND_TYPE|]
+        javaStructFieldDescriptorInitTypeExpr (BT_UserDefined s@Struct {..} params) = [lt|#{structFieldDescriptorInitStructExpr java (qualifiedDeclaredTypeName java s) params}|]
+        javaStructFieldDescriptorInitTypeExpr (BT_UserDefined s@Forward {..} params) = [lt|#{structFieldDescriptorInitStructExpr java (qualifiedDeclaredTypeName java s) params}|]
```

Comment:
Field can be Forward. That's how circular data structures must be represented in Haskell (which lacks pointers). 

---
In reply to: [125132058](https://github.com/Microsoft/bond/pull/517#discussion_r125132058) [](ancestors = 125132058)